Sample/SampleDelegate/SampleDelegate/AppDelegate.swift

```diff
@@ -25,7 +25,7 @@ import SuperDelegate
 @UIApplicationMain
 class AppDelegate: SuperDelegate, ApplicationLaunched {
     
-    let window = UIWindow()
+    var window: UIWindow? = UIWindow()
```

Comment:
we'd need `window` to be `public` to solve this problem, right?

I personally _hate_ making things optional when they don't need to be. Maybe the right answer here is to only initialize `window` in `loadInterface`, since we don't really need one until we have to load our interface — we're likely using a dash more memory than we need to when apps are loaded in the background right now.

The upside of setting `window` in `loadInterface` is that we'd get rid of the `preconditionFailure`:
```
func loadInterface(launchItem: LaunchItem) {
          // Setup our main window since we don't have a storyboard.
          let window = UIWindow()
          self.window = window
          setup(mainWindow: window)
          window.rootViewController = ViewController()
```

Reply:
It does not need to be `public`. If you would have the following scenario: `MyAppDelegate` inherits from `AbstractAppDelegate` which inherits from `SuperDelegate`, you would need this property to be `public`.

Reply:
You're right re public! My first look at this I hadn't realized you were making changes in the sample project :)